HHS/Head-Start-TTADP
# [TTAHUB-1641] Auto generate logical diagram

## Description of change
To remove the manual process of maintaining the logical data model, it has now been automatically generated. This is done either by running yarn ldm or yarn docker:ldm. It is also automatically by running migrations with yarn db:migrate or yarn docker:db:migrate 

In audiation to automatically generating the model, the process identifies discrepancies between the implemented database and the implemented models. When discrepancies are identified, it highlights them and places a issue description in the file. A test is implemented to aid in reporting these discrepancies do CI will fail when they are present.

The generation of the png/svg has been removed from this process and will be solved via another JIRA/method. Only the encoded and plantuml files are generated.

## How to test
Make a change to either the database or models, then run the ldm tool either directly or by running migration. Note that the ldm files under docs will have been modified.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1641


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
